subprojects/snapshots/src/main/java/org/gradle/internal/vfs/watch/FileWatcherRegistry.java

```diff
@@ -36,6 +41,21 @@
         INVALIDATE
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @throws WatchingNotSupportedException when the native watchers can't be updated.
```

Comment:
This should throw something different I think.

Reply:
I think I wrote the wrong Javadoc. The exception is thrown on Linux when we try to what the same path via a symlink. So I suppose the right Javadoc would be:
```
@throws WatchingNotSupportedException when a location can't be watched.
```

We don't throw this exception anywhere else.